lib/packwerk/file_processor.rb

```diff
@@ -16,24 +18,42 @@ def initialize(node_processor_factory:, parser_factory: nil)
       @parser_factory = parser_factory || Packwerk::Parsers::Factory.instance
     end
 
+    sig do
+      params(file_path: String).returns(
+        T::Array[
+          T.any(
+            Packwerk::Reference,
+            Packwerk::Offense,
+          )
+        ]
+      )
+    end
     def call(file_path)
-      parser = @parser_factory.for_path(file_path)
-      return [UnknownFileTypeResult.new(file: file_path)] if parser.nil?
+      return [UnknownFileTypeResult.new(file: file_path)] if parser(file_path).nil?
 
-      node = File.open(file_path, "r", external_encoding: Encoding::UTF_8) do |file|
-        parser.call(io: file, file_path: file_path)
-      rescue Parsers::ParseError => e
-        return [e.result]
-      end
+      node = convert_to_ast_node(file_path)
+      return [] unless node
+
+      references = []
+      node_processor = @node_processor_factory.for(filename: file_path, node: node)
+      node_visitor = Packwerk::NodeVisitor.new(node_processor: node_processor)
+      node_visitor.visit(node, ancestors: [], result: references)
 
-      result = []
-      if node
-        node_processor = @node_processor_factory.for(filename: file_path, node: node)
-        node_visitor = Packwerk::NodeVisitor.new(node_processor: node_processor)
+      references
+    rescue Parsers::ParseError => e
+      [e.result]
+    end
+
+    private
 
-        node_visitor.visit(node, ancestors: [], result: result)
+    def convert_to_ast_node(file_path)
```

Comment:
The name is not wrong, but as the node also contains children, I think it makes more sense to talk about a tree than a single node.

So, just `convert_to_ast` would be a better fit IMO.

Or even `parse_into_ast` to be more specific.